realtime updates for filetree

saurabhsood91 opened the pull request:
Realtime updates for Filetree

zebulasampedro commented on sandstone/filewatcher.py:
```diff
@@ -0,0 +1,58 @@
+from watchdog.observers import Observer
```
`filewatcher.py` should be moved to `lib`, or a submodule within `lib`.

zebulasampedro commented on sandstone/filewatcher.py:
```diff
@@ -0,0 +1,58 @@
+        """
+        Event Handler when a file is created
+        """
+        key = 'filetree:created_new_file'
```
A minor thing, but renaming `created_new_file` to `created_file` would be more consistent.

zebulasampedro commented on sandstone/filewatcher.py:
```diff
@@ -0,0 +1,58 @@
+    _watches = {}
+    _observer = Observer()
+    _event_handler = FilesystemEventHandler()
+    _observer.start()
```
I'd chain `start()` onto line 39 if you can.

saurabhsood91 replied:
The `start()` method doesn't return an instance of `Observer`. I doubt we can chain the methods

zebulasampedro commented on sandstone/lib/tests/python/test_filewatcher.py:
```diff
@@ -0,0 +1,66 @@
+    def setUp(self):
+        self.test_dir = tempfile.mkdtemp()
+
+    def tearDown(self):
```
`tearDown()` should also remove the temp directory created in `setUp()`.

zebulasampedro commented on sandstone/lib/tests/python/test_filewatcher.py:
```diff
@@ -0,0 +1,66 @@
+        Filewatcher.add_directory_to_watch(self.test_dir)
+        self.assertEqual(len(Filewatcher._watches), 1)
+
+    def test_remove_directory_to_watch(self):
```
Cascading tests like this should be avoided when possible.

Also, given that `tearDown()` removes all watches, and is called after each test method, I'm not sure this is exercising the intended path.

zebulasampedro commented on sandstone/lib/tests/python/test_filewatcher.py:
```diff
@@ -0,0 +1,66 @@
+    @mock.patch('sandstone.filewatcher.BroadcastManager.broadcast')
+    def test_file_deleted_event(self, mock_broadcast):
+        # create temp file
+        filepath = os.path.join(self.test_dir, 'tmp.txt')
```
Again, avoid cascading tests.

zebulasampedro commented on sandstone/lib/tests/python/test_filewatcher.py:
```diff
@@ -0,0 +1,66 @@
+import copy
+
+
+class FilewatcherTestCase(unittest.TestCase):
```
This Test Case should validate functionality with multiple watches added.

saurabhsood91 commented:
@zebulasampedro another look? I updated the tests, removed the cascading effect on tests, moved `Filewatcher` to `lib`, and added the test to check for adding multiple watches.

zebulasampedro commented on sandstone/lib/tests/python/test_filewatcher.py:
```diff
@@ -0,0 +1,78 @@
+        for key, value in dirs.items():
+            Filewatcher.remove_directory_to_watch(key)
+        # remove temp directory
+        PosixFS.delete_file(self.test_dir)
```
Use `shutil` or something similar to clean up after the test methods, instead of a component of the software under test.

saurabhsood91 replied:
done
